### ISSUE TITLE
Not support sending empty message

### ISSUE THREAD

**Issue by formulahendry** [Azure/azure-iot-sdk-node#400]
# Context
- **OS and version used:** Windows 10
- **Node.js version:** 10
- **list of installed packages:** 
azure-iot-device-mqtt@1.7.3

# Code sample exhibiting the issue:
```javascript
'use strict';
var connectionString = '***'

var Mqtt = require('azure-iot-device-mqtt').Mqtt;
var DeviceClient = require('azure-iot-device').Client;
var Message = require('azure-iot-device').Message;

var client = DeviceClient.fromConnectionString(connectionString, Mqtt);

function printResultFor(op) {
  return function printResult(err, res) {
    if (err) console.log(op + ' error: ' + err.toString());
    if (res) console.log(op + ' status: ' + res.constructor.name);
  };
}

setInterval(function(){
  var message = new Message("");
  console.log('Sending message: ' + message.getData());
  client.sendEvent(message, printResultFor('send'));
}, 1000);
```

# Console log of the issue:
```
Sending message: 
Sending message: 
e:\Try\Test\node_modules\azure-iot-device-mqtt\node_modules\azure-iot-mqtt-base\lib\mqtt_base.js:169
            throw new ReferenceError('Invalid payload');
            ^

ReferenceError: Invalid payload
    at MqttBase.publish (e:\Try\Test\node_modules\azure-iot-device-mqtt\node_modules\azure-iot-mqtt-base\lib\mqtt_base.js:169:19)
    at constructor.sendEvent (e:\Try\Test\node_modules\azure-iot-device-mqtt\lib\mqtt.js:265:37)
    at constructor.handle (e:\Try\Test\node_modules\machina\lib\machina.js:613:25)
    at constructor.Fsm.(anonymous function) [as handle] (e:\Try\Test\node_modules\machina\lib\machina.js:466:63)
    at e:\Try\Test\node_modules\azure-iot-device-mqtt\lib\mqtt.js:118:44
    at constructor._onEnter (e:\Try\Test\node_modules\azure-iot-device-mqtt\lib\mqtt.js:251:29)
    at constructor.transition (e:\Try\Test\node_modules\machina\lib\machina.js:653:28)
    at constructor.Fsm.(anonymous function) [as transition] (e:\Try\Test\node_modules\machina\lib\machina.js:466:63)
    at e:\Try\Test\node_modules\azure-iot-device-mqtt\lib\mqtt.js:231:56
    at constructor._onEnter (e:\Try\Test\node_modules\azure-iot-device-mqtt\node_modules\azure-iot-mqtt-base\lib\mqtt_base.js:85:25)
```

We could send empty message from C# SDK ans seems Python SDK also support: https://github.com/Azure/azure-iot-sdk-python/issues/179

**Comment by anthonyvercolano** [comment]
Fixed in #401

**Comment by az-iot-builder-01** [comment]
@formulahendry, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
